src/Controllers/Admin/UserWorkLogController.php

```diff
@@ -0,0 +1,162 @@
+    protected $response;
+
+    /** @var User */
+    protected $user;
+
+    /** @var array */
+    protected $permissions = [
+        'admin_user_worklog',
+    ];
+
+    /**
+     * @param Authenticator   $auth
+     * @param Config          $config
+     * @param LoggerInterface $log
+     * @param Worklog         $worklog
+     * @param Redirector      $redirector
+     * @param Response        $response
+     * @param User            $user
+     */
+    public function __construct(
+        Authenticator $auth,
+        Config $config,
+        LoggerInterface $log,
+        Worklog $worklog,
+        Redirector $redirector,
+        Response $response,
+        User $user
+    ) {
+        $this->auth = $auth;
+        $this->config = $config;
+        $this->log = $log;
+        $this->worklog = $worklog;
+        $this->redirect = $redirector;
+        $this->response = $response;
+        $this->user = $user;
+    }
+
+    /**
+     * @param Request $request
+     * @return Response
+     */
+    public function editWorklog(Request $request): Response
+    {
+        $user_id = $request->getAttribute('id');
+        $user = $this->user->findOrFail($user_id);
+
+        $worklog_id = $request->getAttribute('worklog_id');
+        if (isset($worklog_id)) {
+            $worklog = $this->worklog->findOrFail($worklog_id);
+
+            if ($worklog->user->id != $user_id) {
+                throw new HttpNotFound('worklog.not.associated.to.user');
+            }
+            return $this->showEditWorklog($user, $worklog->worked_at, $worklog->hours, $worklog->comment, true);
+        } else {
+            return $this->showEditWorklog($user, $this->getWorkDateSuggestion());
+        }
+    }
+
+    /**
+     * @param Request $request
+     * @return Response
+     */
+    public function saveWorklog(Request $request): Response
+    {
+        $user_id = $request->getAttribute('id');
+        $user = $this->user->findOrFail($user_id);
+
+        $data = $this->validate($request, [
+            'work_date'  => 'required|date:Y-m-d',
+            'work_hours' => 'float|min:0',
+            'comment'    => 'required|max:200',
+        ]);
+
+        $worklog_id = $request->getAttribute('worklog_id');
+        if (isset($worklog_id)) {
+            $worklog = $this->worklog->findOrFail($worklog_id);
+
+            if ($worklog->user->id != $user_id) {
+                throw new HttpNotFound('worklog.not.associated.to.user');
+            }
+        } else {
+            $worklog = new Worklog();
+            $worklog->user()->associate($user);
+            $worklog->creator()->associate($this->auth->user());
+        }
+        $worklog->worked_at = $data['work_date'];
+        $worklog->hours = $data['work_hours'];
+        $worklog->comment = $data['comment'];
+        $worklog->save();
+
+        $this->addNotification(isset($worklog_id) ? 'worklog.edit.success' : 'worklog.add.success');
+
+        return $this->redirect->back();
```

Comment:
Redirecting to the user page might be more useful here but i'm undecided

Reply:
I took it from the UserShirtController, if it helps.

Reply:
That's a good point.
Okay, I will then make it redirect to the user page.
I think it makes sense to unify the structure of all of these tabs at some point – some do have a back button, some don't. On some it shows it as a "<" symbol, on some it is a "back".

Reply:
Thats part of a "Make it more useable" issue -> maybe i should create one ^^